src/preload/preload.ts

```diff
@@ -162,13 +160,13 @@ contextBridge.exposeInMainWorld("misc", miscApi);
 function request<C extends GetCommandIds<"user", "server", "request">>(
     ...args: GetCommandData<GetCommands<"user", "server", "request", C>> extends never ? [commandId: C] : [commandId: C, data: GetCommandData<GetCommands<"user", "server", "request", C>>]
 ): Promise<GetCommands<"server", "user", "response", C>> {
-    return ipcRenderer.invoke("tachyon:request", ...args);
+    return ipcRenderer.invoke("tachyon:request", ...args) as Promise<GetCommands<"server", "user", "response", C>>;
 }
 
 function onEvent<C extends GetCommandIds<"server", "user", "event">>(eventID: C, callback: (event: GetCommandData<GetCommands<"server", "user", "event", C>>) => void) {
     return ipcRenderer.on("tachyon:event", (_event, event) => {
-        if (event.commandId === eventID) {
-            callback(event.data);
+        if (event.commandId === eventID && "data" in event) {
```

Comment:
Possibly not necessary to check for "data" key here but it's not guaranteed to be there by Typescript so it can't hurt to guard against it and fix the error.